src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs

```diff
@@ -6140,6 +6354,106 @@ static C<T> GetC<T>(T x)
                 );
         }
 
+        [ConditionalFact(typeof(MonoOrCoreClrOnly))]
+        public void NullableAnalysis_02()
+        {
+            var src = @"
+#nullable enable
+
+class Program
+{
+    static void M1(Buffer10<string?> b)
+    {
+        System.Span<string> x = b;
+        System.ReadOnlySpan<string> y = b;
+    }
+
+    static void M2(Buffer10<string> b2)
+    {
+        System.Span<string> x = b2;
+        System.ReadOnlySpan<string> y = b2;
+    }
+
+    static void M3(Buffer10<string?> b3)
+    {
+        System.Span<string?> x = b3;
+        System.ReadOnlySpan<string?> y = b3;
+    }
+
+    static void M4(Buffer10<string> b4)
+    {
+        System.Span<string?> x = b4;
+        System.ReadOnlySpan<string?> y = b4;
+    }
+}
+";
+            var comp = CreateCompilation(src + Buffer10Definition, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics(
+                // (8,33): warning CS8619: Nullability of reference types in value of type 'Buffer10<string?>' doesn't match target type 'Span<string>'.
+                //         System.Span<string> x = b;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "b").WithArguments("Buffer10<string?>", "System.Span<string>").WithLocation(8, 33),
+                // (9,41): warning CS8619: Nullability of reference types in value of type 'Buffer10<string?>' doesn't match target type 'ReadOnlySpan<string>'.
+                //         System.ReadOnlySpan<string> y = b;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "b").WithArguments("Buffer10<string?>", "System.ReadOnlySpan<string>").WithLocation(9, 41),
+                // (26,34): warning CS8619: Nullability of reference types in value of type 'Buffer10<string>' doesn't match target type 'Span<string?>'.
+                //         System.Span<string?> x = b4;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "b4").WithArguments("Buffer10<string>", "System.Span<string?>").WithLocation(26, 34),
+                // (27,42): warning CS8619: Nullability of reference types in value of type 'Buffer10<string>' doesn't match target type 'ReadOnlySpan<string?>'.
+                //         System.ReadOnlySpan<string?> y = b4;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "b4").WithArguments("Buffer10<string>", "System.ReadOnlySpan<string?>").WithLocation(27, 42)
```

Comment:
Should we test something similar separately for nullable value types (like `int?`)?

Reply:
> Should we test something similar separately for nullable value types (like ```int?```)?

A scenario like that wouldn't be any different than the one tested in```Conversion_ElementTypeMismatch```. That is because ```int?`` and ```int``` are different types, whereas ```string?``` and ```string``` are the same type, but with different nullability.